draft-rha-jose-hpke-encrypt.md

```diff
@@ -220,44 +220,86 @@ This example demonstrates the representaton of an encapsulated key as a JWK.
 
 ### Integrated Encryption
 
-This mode only supports a single recipient and JWE Compact serialization.
+This mode only supports a single recipient.
 
 HPKE is employed to directly encrypt the plaintext, and the resulting ciphertext is included in the JWE ciphertext. 
 
-The sender MUST specify the 'epk' and 'alg' parameters in the protected header to indicate the use of HPKE. 
+In Integrated Encryption mode:
 
-The 'enc' (Encryption Algorithm) parameter MUST NOT be present. 
+*  The "epk" Header Parameter MUST be present, it MUST contain an Encapsulated JSON Web Key and it MUST occur only within the JWE Protected Header.
 
-This is a deviation from the rule in Section 4.1.2 of {{RFC7516}}. 
+*  The "alg" Header Parameter MUST be "HPKE-IntEnc", "enc" MUST be an HPKE algorithm from JSON Web Signature and Encryption Algorithms in {{JOSE-IANA}} and they MUST occur only within the JWE Protected Header.
 
-In Integrated Encryption mode:
+*  The JWE Ciphertext MUST be the base64url encoded 'ct' value.
```

Comment:
```suggestion
*  The JWE Ciphertext MUST be the resulting HPKE ciphertext and tag ('ct' value) encoded using base64url.
```

Reply:
I propose the following text:
The JWE Ciphertext MUST be the resulting HPKE ciphertext ('ct' value) encoded using base64url. For example, if an AES-GCM scheme is used, the GCM authentication tag is placed at the end of the HPKE ciphertext output.

Reply:
That's confusing because HPKE produces only ciphertext and tag `ct`, that's one output value, not two. And the internals of HPKE shouldn't be plucked out of HPKE and placed in examples here.

Reply:
We probably want to say something like this once:

`When tag is non empty, it is the base64url encoded bytes of the AEAD used`

Reply:
I don't get it. JWE Authentication Tag will always be empty with HPKE.

Reply:
> That's confusing because HPKE produces only "ciphertext and tag `ct`", that's one output value, not two. And the internals of HPKE shouldn't be plucked out of HPKE and placed in examples here.

Yes, It is implied from HPKE that ct includes both ciphertext and authentication tag. I will modify the text as follows:
The JWE Ciphertext MUST be the resulting HPKE ciphertext ('ct' value) encoded using base64url.

Reply:
Exactly. All my suggestion is trying to do is to assign as source and name the HPKE "ct" output as per HPKE's own language.

> The JWE Ciphertext MUST be the base64url encoded 'ct' value.

> The JWE Ciphertext MUST be the resulting HPKE ciphertext and tag ('ct' value) encoded using base64url.

Reply:
I removed the "For example,..." line.